### PR TITLE
Add a new method to permit creation of buckets from the bucketset

### PR DESCRIPTION
**Pull request by vagababov**
This is a natural factory.

I don't want to mess with Yanwei's PR, so I didn't remove the old method.
But we can reconcile that in the end.

/assign @yanweiguo @evankanderson

### CODE REVIEW

File: hash/bucketer.go [hunk 1]
```diff
@@ -86,36 +86,44 @@ func (b *Bucket) Has(nn types.NamespacedName) bool {
 	return b.buckets.Owner(nn.String()) == b.name
 }
 
+// NewBucket creates a new Bucket with given name based on this bucketset.
+func (bs *BucketSet) NewBucket(name string) *Bucket {
+	return &Bucket{
+		name:    name,
```
Comment (yanweiguo) [inline hash/bucketer.go:92]: Since you're here, I think it's chance to add a check that the given name is in the bucketSet.buckets.

Comment (evankanderson) [reply hash/bucketer.go:92]: +1

Comment (vagababov) [reply hash/bucketer.go:98]: Sounds good!
Done.

**Pull request merged**